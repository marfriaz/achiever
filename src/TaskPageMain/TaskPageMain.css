.TaskPageMain {
  background-color: #efefef;
  height: 100%;
  padding: 10px;
  width: 100%;
}

.TaskPageMain__content {
  color: white;
  max-width: 380px;
}
