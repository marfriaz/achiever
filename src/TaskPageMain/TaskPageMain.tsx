import React, { useContext } from "react";
import { useHistory, useParams } from "react-router-dom";
import Task from "../Task/Task";
import ApiContext from "../ApiContext";
import { findTask } from "../tasks-helpers";
import "./TaskPageMain.css";

export default function TaskPageMain() {
  interface ParamTypes {
    taskId: string;
  }
  const { taskId } = useParams<ParamTypes>();

  const history = useHistory();

  const handleDeleteTask = (taskId: number) => {
    history.push(`/`);
  };

  const { tasks } = useContext(ApiContext)!;
  const task = findTask(tasks, taskId) || {
    id: 0,
    name: "",
    details: "",
    date_created: `${new Date()}`,
  };
  return (
    <section className="TaskPageMain">
      <Task
        id={task.id}
        name={task.name}
        modified={task.date_created}
        onDeleteTask={handleDeleteTask}
      />
      <div className="TaskPageMain__content">
        {task.details.split(/\n \r|\n/).map((para: any, i: any) => (
          <p key={i}>{para}</p>
        ))}
      </div>
    </section>
  );
}
