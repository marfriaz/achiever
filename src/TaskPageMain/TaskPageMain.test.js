import React from "react";
import { shallow } from "../GoalListNav/node_modules/enzyme";
import toJson from "../GoalListNav/node_modules/enzyme-to-json";
import TaskPageMain from "./TaskPageMain";

describe(`TaskPageMain component`, () => {
  it("renders a .TaskPageMain by default", () => {
    const wrapper = shallow(<TaskPageMain />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  // enzyme doesn't yet support React.createContext
  it.skip("renders a Task with task prop", () => {
    const props = {
      match: {
        params: {
          taskId: "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        },
      },
    };
    const context = {
      tasks: [
        {
          id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
          name: `Dogs`,
          modified: `2019-01-03T00:00:00.000Z`,
          // goalId: b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1,
          content: "Corporis accusamus placeat.\n \rUnde.",
        },
      ],
    };
    const task = shallow(<TaskPageMain {...props} />, context).find("Task");
    expect(toJson(task)).toMatchSnapshot();
  });

  // enzyme doesn't yet support React.createContext
  it.skip(`splits the content by \\n or \\n\\r, with a p foreach`, () => {
    const props = {
      match: {
        params: {
          taskId: "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
        },
      },
    };

    const tasksContextWithDifferentContent = [
      {
        tasks: [
          {
            id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
            content: "Content with n r.\n \rafter n r.",
          },
        ],
      },
      {
        tasks: [
          {
            id: `cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1`,
            content: "Content with n.\nafter.",
          },
        ],
      },
    ];

    tasksContextWithDifferentContent.forEach((context) => {
      const content = shallow(<TaskPageMain {...props} />, context).find(
        "TaskPageMain__content"
      );
      expect(toJson(content)).toMatchSnapshot();
    });
  });
});
