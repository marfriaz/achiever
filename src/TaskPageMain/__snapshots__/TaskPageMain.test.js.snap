// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskPageMain component renders a .TaskPageMain by default 1`] = `
<section
  className="TaskPageMain"
>
  <Task />
  <div
    className="TaskPageMain__content"
  >
    <p
      key="0"
    />
  </div>
</section>
`;

exports[`TaskPageMain component renders a Task with task prop 1`] = `
<Task
  id="cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1"
  modified="2019-01-03T00:00:00.000Z"
  name="Dogs"
/>
`;

exports[`TaskPageMain component splits the content by \\n or \\n\\r, with a p foreach 1`] = `null`;

exports[`TaskPageMain component splits the content by \\n or \\n\\r, with a p foreach 2`] = `null`;
